src/main/java/com/direwolf20/buildinggadgets/common/tools/InventoryManipulation.java

```diff
@@ -221,7 +230,12 @@ public static int countItem(ItemStack itemStack, EntityPlayer player, IRemoteInv
 
         if (invContainers.size() > 0) {
             for (IItemHandler container : invContainers) {
-                count += countInContainer(container, itemStack.getItem(), itemStack.getMetadata());
+            	{
+            		if(container instanceof IBlockProviderItemHandler)
+            			count += ((IBlockProviderItemHandler) container).getItemCount(itemStack, player);
+            		else
+            			count += countInContainer(container, itemStack.getItem(), itemStack.getMetadata());
```

Comment:
My suggestion would allow replacing this again with wrapping the container in an adapter and just passing simulate==false to the extractItems Method.